### DISCUSSION TITLE
Displacement issues occurring after rendering with TTF fonts

### DISCUSSION THREAD

**Discussion by inernoro** [SixLabors/Fonts#373] [General] [answered]
> It seems that only the letter 'a' is problematic. I'm not sure if it's an issue with my font or something else. Other fonts also exhibit similar issues to varying degrees, such as offsetting, spacing, and inability to center. 

```
[Fact]
    public void FontTest2()
    {
        var size = 200;
        var rgb32 = GetSquareRgba32(size, Color.White);
        var fontFamily = new FontCollection().Add("/inernoro/vol/Genshin default fonts.ttf");
        var font = new Font(fontFamily, 50);
        var op = new RichTextOptions(font)
        {
            HorizontalAlignment = HorizontalAlignment.Center,
            TextAlignment = TextAlignment.Center,
            VerticalAlignment = VerticalAlignment.Center,
            Origin = new Vector2(size, size) / 2
        };
        rgb32.Mutate(w =>
        {
             w.DrawText(op, "A", Color.Black);
             w.DrawText(op, "B", Color.Blue);
        });
        rgb32.Save(AsPng());
    }
```
![45bc8ef17ecc4db5bde938d](https://github.com/SixLabors/Fonts/assets/9151851/6110a27c-7b59-4032-bd2e-e1ac7b78dde3) 

> other test

![f9943b587ade46909efe239](https://github.com/SixLabors/Fonts/assets/9151851/7c16d7b7-03b5-4394-a07d-7f8972b72d9c) ![0e29f303a71542d8822ff7a](https://github.com/SixLabors/Fonts/assets/9151851/f2cae471-62e1-4425-ad02-b8e7b6bb457e) ![547b4e2ccb3940a88bf6234](https://github.com/SixLabors/Fonts/assets/9151851/820ed86f-1e31-4c88-ab73-b75e63d46cca)

[Genshin default fonts.zip](https://github.com/SixLabors/Fonts/files/13751147/Genshin.default.fonts.zip)

> windows 
![image](https://github.com/SixLabors/Fonts/assets/9151851/29cbb2b5-d911-4350-a03d-cc9de16fb10c)

<img src="https://github.com/SixLabors/Fonts/assets/9151851/7e15add4-da36-47b2-8ebf-584c39c2120d" alt="image" width="50%" height="50%">

**Comment by JimBobSquarePants** [comment, marked as answer]
Tuns out an old hack that was designed to work around negative top side bearings leading to cutoff (_the proper fix is to use the correct line spacing for the font_) was causing this.

I've removed it as part of #380 

![AB](https://github.com/SixLabors/Fonts/assets/385879/a2c84f7c-423f-4b15-8217-2d567912b9c3)

**Reply by JimBobSquarePants** [reply]
Fix is released.
